src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,207 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+class spell_pri_prayer_of_mending_aura : public SpellScriptLoader
+{
+    public:
+        spell_pri_prayer_of_mending_aura() : SpellScriptLoader("spell_pri_prayer_of_mending_aura") { }
+
+        class spell_pri_prayer_of_mending_aura_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_pri_prayer_of_mending_aura_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_HEAL });
+            }
+
+            void HandleHeal(ProcEventInfo & /*eventInfo*/)
+            {
+                // Caster: the player(priest) that cast the prayer of mending
+                // Target: the player that currently has the prayer of mending aura on him
+                Unit* target = GetTarget();
+                if (Unit* caster = GetCaster())
+                {
+                    // Cast the spell to heal the owner
+                    caster->CastSpell(target, SPELL_PRIEST_PRAYER_OF_MENDING_HEAL);
+
+                    // Only cast jump if stack is higher than 0
+                    int32 stackAmount = GetStackAmount();
+                    if (stackAmount > 1)
+                        target->CastCustomSpell(SPELL_PRIEST_PRAYER_OF_MENDING_JUMP, SPELLVALUE_BASE_POINT0, stackAmount - 1, target, true, nullptr, nullptr, caster->GetGUID());
+                    Remove();
+                }
+            }
+
+            void Register() override
+            {
+                OnProc += AuraProcFn(spell_pri_prayer_of_mending_aura_AuraScript::HandleHeal);
+            }
+        };
+
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_pri_prayer_of_mending_aura_AuraScript();
+        }
+};
+
+// 155793 - prayer of mending (Jump) - SPELL_PRIEST_PRAYER_OF_MENDING_JUMP
+class spell_pri_prayer_of_mending_jump : public SpellScriptLoader
 {
     public:
-        spell_pri_prayer_of_mending_heal() : SpellScriptLoader("spell_pri_prayer_of_mending_heal") { }
+        spell_pri_prayer_of_mending_jump() : SpellScriptLoader("spell_pri_prayer_of_mending_jump") { }
 
-        class spell_pri_prayer_of_mending_heal_SpellScript : public SpellScript
+        class spell_pri_prayer_of_mending_jump_SpellScript : public spell_pri_prayer_of_mending_SpellScriptBase
         {
-            PrepareSpellScript(spell_pri_prayer_of_mending_heal_SpellScript);
+            PrepareSpellScript(spell_pri_prayer_of_mending_jump_SpellScript);
+
+            enum class TargetPriority : uint8
+            {
+                Players,
+                Pets,
+                None,
+            };
 
-            void HandleHeal(SpellEffIndex /*effIndex*/)
+            void OnTargetSelect(std::list<WorldObject*>& targets)
             {
-                if (Unit* caster = GetOriginalCaster())
+                // find best target
+                // Make sure too exclude the player that had the aura (caster in our case)
```

Comment:
Cosmetic/grammar comment details:
```suggestion
                // Find the best target.
                // Make sure to exclude player wearing the aura (caster in this case)
```